tools/libboard/board_dc/board_dc.c

```diff
@@ -326,3 +340,126 @@ fpga_result print_board_info(fpga_token token)
 
 	return resval;
 }
+
+// Read sysfs
+fpga_result read_sysfs(fpga_token token, char *sysfs_path, char *sysfs_name)
+{
+	fpga_result res = FPGA_OK;
+	fpga_result resval = FPGA_OK;
+	uint32_t size = 0;
+	char name[FPGA_STR_SIZE] = { 0 };
+	fpga_object sec_object;
+
+	if (sysfs_path == NULL ||
+		sysfs_name == NULL) {
+		OPAE_ERR("Invalid input parameter");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = fpgaTokenGetObject(token, sysfs_path, &sec_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get token Object");
+		return res;
+	}
+
+	res = fpgaObjectGetSize(sec_object, &size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get object size ");
+		resval = res;
+		goto out_destroy;
+	}
+
+	if (size > FPGA_STR_SIZE) {
+		OPAE_ERR("object size bigger then buffer size");
+		resval = FPGA_EXCEPTION;
+		goto out_destroy;
+	}
+
+	res = fpgaObjectRead(sec_object, (uint8_t *)(&name), 0, size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to Read object ");
+		resval = res;
+		goto out_destroy;
+	}
+
+	snprintf(sysfs_name, FPGA_STR_SIZE, "%s", (char *)name);
+
+out_destroy:
+	res = fpgaDestroyObject(&sec_object);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to Destroy Object");
+		resval = res;
+	}
+
+	return resval;
+}
+
+
+fpga_result print_sec_info(fpga_token token)
+{
+	fpga_result res = FPGA_OK;
+	fpga_object tcm_object;
+	char name[FPGA_STR_SIZE] = { 0 };
+
+	res = fpgaTokenGetObject(token, DFL_SYSFS_SEC_GLOB, &tcm_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_MSG("Failed to get token Object");
+		return res;
+	}
+	printf("********** SEC Info START ************ \n");
+
+	// BMC Keys
+	memset(name, 0, sizeof(name));
+	if (read_sysfs(token, DFL_SYSFS_SEC_BMC_ROOT, name) == FPGA_OK)
+		printf("BMC root entry hash: %s", name);
+	else
```

Comment:
This function returns an fpga_result.  Shouldn't these failed calls to read_sysfs affect the return value?

Reply:
fpgainfo out prints shows lot of error messages, so ignore errors.

Reply:
What is the process exit code when an error occurs?

Reply:
fixed , exits with error code